Infrastructure/MetricLogging/LoggingTuitionPartnerService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Diagnostics;
+using Application;
+using Domain.Enums;
+using Domain.Search;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.MetricLogging;
+
+public class LoggingTuitionPartnerService : ITuitionPartnerService
+{
+    private readonly ITuitionPartnerService _inner;
+    private readonly ILogger _logger;
+
+    public LoggingTuitionPartnerService(ITuitionPartnerService inner, ILogger<LoggingTuitionPartnerService> logger)
+        => (_inner, _logger) = (inner, logger);
+
+    public async Task<int[]?> GetTuitionPartnersFilteredAsync(TuitionPartnersFilter filter, CancellationToken cancellationToken)
+    {
+        using var _ = _logger.BeginScope("{@Filter}", filter);
+        _logger.LogInformation("Filtering tuition partners");
+
+        var stopwatch = new Stopwatch();
+        stopwatch.Start();
+        var result = await _inner.GetTuitionPartnersFilteredAsync(filter, cancellationToken);
+        stopwatch.Stop();
+
+        var resultCount = result?.Length;
+
+        var logLevel = (resultCount == 0 && string.IsNullOrWhiteSpace(filter.Name)) ? LogLevel.Warning : LogLevel.Information;
```

Comment:
What's the reason that we might log this case as a warning rather than as info?

Reply:
This is lifted from the existing "SearchTuitionPartnerHandlerLoggingBehaviour.cs" - I guess at the moment there should be no filters possible (other than Name) that would return 0 results.  So if they do return 0 it's probably an issue, something has gone wrong.  But if we add any new filters in the future and it results in 0 results on a regular basis then I think we'd want it to switch to info.